include/HubFramework/HUBComponentLayoutManager.h

```diff
@@ -112,6 +114,18 @@ typedef NS_ENUM(NSUInteger, HUBComponentLayoutContentEdge) {
 - (CGFloat)horizontalOffsetForComponentsWithLayoutTraits:(NSArray<NSSet<HUBComponentLayoutTrait> *> *)componentsTraits
                    firstComponentLeadingHorizontalOffset:(CGFloat)firstComponentLeadingOffsetX
                    lastComponentTrailingHorizontalOffset:(CGFloat)lastComponentTrailingOffsetX;
+
+/**
+ *  Return top margin for overlay component with the specified model
+ *
+ *  @param componentModel The component model for the overlay component
+ *
+ *  @discussion If you want the overlay component to be centered top margin should be 0.
+ *  In case you want it to be shifted down, e.g. to reveal header component, return corresponding top margin
+ *  from this delegate method.
+ */
+- (CGFloat)topMarginForOverlayComponentWithModel:(id<HUBComponentModel>)componentModel;
```

Comment:
We should expose the component model itself to the layout manager. That will quickly lead to very hard-coded messy rules around layout. Instead, we should use layout traits to make all layout decisions. This scales very well across a large code base, and enables more succinct rules to be defined. See the [Layout programming guide](https://spotify.github.io/HubFramework/layout-programming-guide.html) for more context. Feel free to introduce a new layout trait if you need to to accomplish your task 🙂